### PR TITLE
cli/args: Make build_from_source public instance method

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
#5730

### CODE REVIEW

**Comment by GauthamGoli** [comment]
@MikeMcQuaid  Some tests are failing whenever `install_test_formula` is being called. One way to fix them is by adding a `nil` check on `cmdline_args` whenever its being used in `cli/args.rb` but I don't think thats the right way, instead, may be have a generic way of initiating args in tests. 

Need your input on how to solve this.

File: Library/Homebrew/cli/parser.rb [hunk 1]
```diff
@@ -259,7 +259,7 @@ def common_switch(name)
       end
 
       def option_passed?(name)
-        @args.respond_to?(name) || @args.respond_to?("#{name}?")
+        @args[name.to_sym] || @args["#{name}?".to_sym]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cli/parser.rb:262]: Why this change?

Comment (GauthamGoli) [reply Library/Homebrew/cli/parser.rb:262]: `build_from_source` is an instance method now and `respond_to?` would always return `true`.

**Comment by GauthamGoli** [comment]
This is a cleaner approach. Thanks for the help @MikeMcQuaid

**Comment by MikeMcQuaid** [comment]
You're welcome @GauthamGoli, thanks for the PR!
